File: LeetCode_1018/Main.js
/**
 * @param {number[]} nums
 * @return {boolean[]}
*/
var prefixesDivBy5 = function (nums) {
    const answer = [];
    let prefix = 0;
    for (let num of nums) {
        prefix = ((prefix << 1) + num) % 5;
        answer.push(prefix === 0);
    }
    return answer;
};

File: LeetCode_1018/Main.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<bool> prefixesDivBy5(vector<int>& nums) {
        int prefix = 0;
        vector<bool> answer;

        for (int num : nums) {
            prefix = ((prefix * 2) + num) % 5;
            answer.push_back(prefix == 0);
        }
        return answer;
    }
};